syntax = "proto3";
import "service_discovery.proto";


message BookScooterRequest {
    string scooter_id = 1;
    string start = 2;
    string user_email = 3;
    string title = 4;
}

message BookScooterResponse {
    int64 id = 1;
    string start = 2;
    string user_email = 3;
    string scooter_id = 4;
    string end = 5;
    string title = 6;
}

message EndRideRequest {
    int64 id = 1;
}

message GetBookingRequest {
    int64 id = 1;
}

message GetAllBookingsResponse {
    repeated BookScooterResponse bookings = 1;
}

service BookingsService {
    rpc BookScooter(BookScooterRequest) returns (BookScooterResponse);
    rpc EndRide(EndRideRequest) returns (BookScooterResponse);
    rpc GetBooking(GetBookingRequest) returns (BookScooterResponse);
    rpc GetAllBookings(Empty) returns (GetAllBookingsResponse);
}

